src/main/java/heronarts/lx/LXComponent.java

```diff
@@ -884,11 +885,19 @@ protected LXComponent addParameter(String path, LXParameter parameter) {
     }
     if (parameter instanceof ColorParameter) {
       // NOTE: order is important, brightness must be saved/loaded first
+      // TODO: Why?
       ColorParameter colorParameter = (ColorParameter) parameter;
       addParameter(path + "/" + ColorParameter.PATH_BRIGHTNESS, colorParameter.brightness);
       addParameter(path + "/" + ColorParameter.PATH_SATURATION, colorParameter.saturation);
       addParameter(path + "/" + ColorParameter.PATH_HUE, colorParameter.hue);
     }
+    if (parameter instanceof LinkableColorParameter) {
```

Comment:
This is the kind of special-casing mess I want to clean up with AggregateParameter... more to come on that soon when I find time...